packages/yarnpkg-fslib/sources/PosixFS.ts

```diff
@@ -11,11 +11,11 @@ export class PosixFS extends ProxiedFS<NativePath, PortablePath> {
     this.baseFs = baseFs;
   }
 
-  protected mapFromBase(path: PortablePath) {
-    return npath.fromPortablePath(path);
+  protected mapFromBase(path: PathLike<PortablePath>) {
+    return npath.fromPortablePath(ppath.fromPathLike(path));
   }
 
-  protected mapToBase(path: NativePath) {
-    return npath.toPortablePath(path);
+  protected mapToBase(path: PathLike<NativePath>) {
+    return npath.toPortablePath(npath.fromPathLike(path));
   }
```

Comment:
Btw, shouldn't we return the same type of `PathLike` as the one we receive? As we saw, the result of `Buffer.from(Buffer.from([0xca, 0xc5, 0x0e]).toString('binary'), 'binary')` is correct while the result of `Buffer.from(Buffer.from([0xca, 0xc5, 0x0e]).toString('binary'))` isn't. That means that the string representation would break if definitely turned into a string, right?

Reply:
It would also be safer to include tests that use `[0xca, 0xc5, 0x0e]` to be sure that we handle this tricky case properly.